### PR TITLE
Add README and tests

### PR DESCRIPTION
**Pull request by ezmiller**
@AnnJohn @thomastuts This is ready for review.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,17 +2,68 @@
...
+
+1. In the `components/DumbComponent.js` file, create a `DumbComponent` component.
+2. The component should be declared as a class (instance of `Component`) so
+   that it can carry state.
+3. The component should have a state property called `mood` that has a default
+   value `happy`.
+4. The component should simply render its current `mood` state to the page in
+   a div.
+5. The component should implement a `handleClick` function that can serve as a
+   callback to the <div>'s click event. When clicked the component's mood
```
Comment (thomastuts) [inline README.md]: `<div>` is missing backticks, causing it to not show up in a rendered Markdown file

File: README.md [hunk 2]
```diff
@@ -2,17 +2,68 @@
...
+   that it can carry state.
+3. The component should have a state property called `mood` that has a default
+   value `happy`.
+4. The component should simply render its current `mood` state to the page in
+   a div.
+5. The component should implement a `handleClick` function that can serve as a
+   callback to the <div>'s click event. When clicked the component's mood 
+   should toggle between happy and sad states.
+
+The fact that this components mood fluctuates when clicked,
```
Comment (thomastuts) [inline README.md]: This **component's**? Just a suggestion, possession is always kind of finicky (as a non-native speaker, anyway)

File: README.md [hunk 3]
```diff
@@ -2,17 +2,68 @@
...
+while part of our design, makes it a more unpredictable part of our UI. As our
+program runs, and users interact with it, we won't be able to predict what
+state our component is in. Obviously, many components need state -- because our
+are UIs are interactive -- but where possible it's good to avoid
+state entirely, as we will in the second component in this lab.
```
Comment (thomastuts) [inline README.md]: we will **see**

File: README.md [hunk 4]
```diff
@@ -2,17 +2,68 @@
...
+4. It should receiving one property called `handleClick` that sends a message
+   to the browser via the `alert` method.
+
+When you've finished this component, take a moment to compare it to the previous
+presentational component we wrote. See how much more stable it is? It has no ability
+to change its output internally. We can always know, based on the props that we
+provide, what type of component it will produce. Note, as well, that this doesn't mean
+that the component lacks interactivity. We can actually determine a wide variety of
+click behaviors on the component just by providing a different callback. It's just that
+the component cannot itself determine its behavior. And this kind of "dumbness" is
```
Comment (thomastuts) [inline README.md]: 'the component cannot itself' -> 'the component itself cannot'? Minor suggestion.

I think we can also remove the 'And' in the second sentence, and maybe call out why it's such a good thing (again): predictable components are much easier to work with (or something along those lines)?

File: test/DumberComponent_test.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  beforeEach(() => {
+    spy = sinon.spy();
+    wrapper = !DumberComponent.prototype ?
+      shallow(<Noop />) : shallow(<DumberComponent handleClick={spy} />);
+  });
+
+  it('should not be a class-based component', () => {
+    const tryToGetState = () => { wrapper.state(); }
+    expect(tryToGetState).toThrow(
+      'ShallowWrapper::state() can only be called on class components'
```
Comment (thomastuts) [inline test/DumberComponent_test.js]: I'd 'humanize' this error output a little more: maybe a simple '<DumberComponent />' should be a regular function, or something similar?

File: README.md [hunk 6]
```diff
@@ -2,17 +2,68 @@
...
+
+Although our `DumbComponent` fits the pattern of a presentation component, the
+fact that it has state makes it unstable. To practice building a more stable
+type of presentational component, let's build a `DumberComponent` to meet the
+following specifications:
+
+1. In the `components/DumberComponent.js` file, create a `DumberComponent` component.
+2. The component should be a "stateless functional" component.
+3. It should render a `<div>` to the page that contains the text: "I am just happy".
+4. It should receiving one property called `handleClick` that sends a message
```
Comment (thomastuts) [inline README.md]: It should **receive**

Also, do we need to be this specific? Maybe we shouldn't be so concerned with what they do with the function when it is called, as long as it's called? We're using a spy in the tests anyway, so the `alert` thing seems kind of redundant here in my opinion.

File: README.md (hunk 2 above)
Comment (ezmiller) [reply README.md]: You are definitely right.

File: test/DumberComponent_test.js (hunk 5 above)
Comment (ezmiller) [reply test/DumberComponent_test.js]: This error output actually comes from enzyme, and the toThrow method is checking for that particular message as a way to ensure that they are getting **that** error and not another. But let me see if I can add a custom message. Another possibility would be to throw a custom error in the `tryToGetState` fn above.

Comment (thomastuts) [reply test/DumberComponent_test.js]: Ah right, sorry, totally missed that. You _should_ be able to add a custom message as the second argument to `.toThrow()` — that's how it works for the other methods like `.toEqual()` etc anyway.

File: README.md (hunk 6 above)
Comment (ezmiller) [reply README.md]: n.b.

**Comment by ezmiller** [comment]
@thomastuts changes made. Want to review and let me know if it's ready for merge. I was able to add the improved error message on the test for the stateless component.

**Review by thomastuts — APPROVED** [review]
@ezmiller This looks great, thanks for the changes! Feel free to merge it in. Can you merge the updates into `wip-solution` too, before merging that branch?

**Comment by ezmiller** [comment]
@thomastuts I updated wip-solution.

**Pull request merged**